### ISSUE TITLE
require-default-props does not detect properties in quotes

### ISSUE THREAD

**Issue by zech** [yannickcr/eslint-plugin-react#1201]
Hi,

I've got an issue with code like this

```javascript
DueEntryComponent.propTypes = {
    "isOverdue": PropTypes.bool
};

DueEntryComponent.defaultProps = {
    "isOverdue": false
};
```

triggers the message:

`propType "isOverdue" is not required, but has no corresponding defaultProp declaration. (react/require-default-props)`

changing the declaration to

```javascript
DueEntryComponent.defaultProps = {
    isOverdue: false
};
```

omits the error.

I don't think quotes should have an effect on detecting defaultProps. Maybe I'm wrong ;-)

**Comment by ljharb** [comment]
While unnecessarily quoted keys isn't my preferred style, I agree this should work. I think it should also work as a computed template literal with no interpolations.

**Comment by yannickcr** [comment]
Fixed it for quoted properties. I will add template literal support in another patch.
